tests/test.py

```diff
@@ -1,36 +1,54 @@
 import time
 import sys
 import os
+import string
+import random
+import itertools
 
 sys.path.append(os.sep.join([os.path.dirname(os.path.realpath(__file__)),
                              '..',
-                             'file_checker'
+                             'src'
                              ]))
 import util # noqa
+from file_checker import FileChecker # noqa
+from enums import FORMATS # noqa
+
+
+def get_error_string(errors):
+    err_str = '\n'
+    for error in errors:
+        err_str += error + '\n\n'
+    return err_str
 
 
 class TestUtil():
     def test_gen_unique_str(self):
         gen_unique_str = util.gen_unique_str
+        errors = []
         u = gen_unique_str('')
 
         s = ''
         if gen_unique_str(s) in s:
-            raise Exception("'{0}' is in '{1}'".format(gen_unique_str(s), s))
+            errors += ["'{0}' is in '{1}'".format(gen_unique_str(s), s)]
         s = u
         if gen_unique_str(s) in s:
-            raise Exception("'{0}' is in '{1}'".format(gen_unique_str(s), s))
+            errors += ["'{0}' is in '{1}'".format(gen_unique_str(s), s)]
 
         # benchmark
         start = time.time()
         for i in range(0, 1000):
             # 256 chars
             gen_unique_str(u * 256)
 
-        assert time.time() - start < 0.5
+        if time.time() - start > 0.5:
+            errors += ['The algorithm is not efficient.']
+
+        if errors:
```

Comment:
I personally don't like the approach of throwing an exception if there are errors. Testing usually involves a series of asserts, and exceptions are meant for _exceptional_ behavior anyways.